### PR TITLE
Update registries sync status after sync from registry action

### PR DESCRIPTION
**Pull request by jerabekjiri**
Issue: [AAH-1094](https://issues.redhat.com/browse/AAH-1094)

UI part of issue AAH-1094. Fixed backend here [#1056](https://github.com/ansible/galaxy_ng/pull/1056) (merged). 

With fixed backend `registry sync status` is now showing correctly, but it doesn't get updated after the `sync from registry` action. We have to manually reload the page to see the result. This PR fixes the status and keeps fetching registries until the status is completed.

### CODE REVIEW

File: src/containers/execution-environment/registry-list.tsx [hunk 1]
```diff
@@ -447,6 +447,22 @@ class ExecutionEnvironmentRegistryList extends React.Component<
     );
   }
 
+  private updateRegistries() {
+    ExecutionEnvironmentRegistryAPI.list(this.state.params).then((result) => {
+      const isAllCompleted = result.data.data.every(
+        (task) => task.last_sync_task.state === 'completed',
```
Comment (himdel) [inline src/containers/execution-environment/registry-list.tsx:453]:
```suggestion
        (task) => ['canceled', 'completed', 'failed'].includes(task.last_sync_task.state),
```

(at least that's what task deail does)

File: src/containers/execution-environment/registry-list.tsx [hunk 2]
```diff
@@ -447,6 +447,22 @@ class ExecutionEnvironmentRegistryList extends React.Component<
...
+      );
+
+      if (!isAllCompleted) setTimeout(() => this.updateRegistries(), 500);
```
Comment (himdel) [inline src/containers/execution-environment/registry-list.tsx:456]: A 500ms interval from everybody viewing the registry list might spam the server...

We're not quite being consistent with the timeouts...

```
src/containers/user-management/delete-user-modal.tsx
88:        await new Promise((r) => setTimeout(r, 500));

src/utilities/wait-for-task.ts
13:      return new Promise((r) => setTimeout(r, 5000)).then(() =>

src/containers/execution-environment-detail/base.tsx
229:            setTimeout(() => this.loadRepo(), 10000);

src/containers/execution-environment-detail/tag-manifest-modal.tsx
353:          await new Promise((r) => setTimeout(r, 5000));

src/containers/certification-dashboard/certification-dashboard.tsx
510:        await new Promise((r) => setTimeout(r, 500));

src/components/repositories/remote-repository-table.tsx
30:    this.polling = setInterval(() => {
...
48-    }, 5000);

src/containers/my-imports/my-imports.tsx
78:    this.polling = setInterval(() => {
...
89-    }, 10000);

src/containers/task-management/task_detail.tsx
405:      this.setState({ refresh: setInterval(() => this.loadContent(), 10000) });
```

but I'd suggest 5000 or 10000 :)

**Comment by himdel** [comment]
I think we should merge `queryRegistries` and `updateRegistries`...

both use `ExecutionEnvironmentRegistryAPI.list`, both set `items`, `itemCount`, `loading`,
but right now, changing the page or reloading the page won't make the list autoupdate, so it works only after pushing the sync button, not when seeing a syncing task.

What about something like...

```diff
diff --git a/src/containers/execution-environment/registry-list.tsx b/src/containers/execution-environment/registry-list.tsx
index b1ea942..8bddf1b 100644
--- a/src/containers/execution-environment/registry-list.tsx
+++ b/src/containers/execution-environment/registry-list.tsx
@@ -434,34 +434,24 @@ class ExecutionEnvironmentRegistryList extends React.Component<
     );
   }
 
-  private queryRegistries() {
-    this.setState({ loading: true }, () =>
-      ExecutionEnvironmentRegistryAPI.list(this.state.params).then((result) =>
+  private queryRegistries(noLoading = false) {
+    this.setState(noLoading ? null : { loading: true }, () =>
+      ExecutionEnvironmentRegistryAPI.list(this.state.params).then((result) => {
+        const isAnyRunning = result.data.data.some((task) => ['running', 'waiting'].includes(task.last_sync_task.state));
+
+        if (isAnyRunning) {
+          setTimeout(() => this.queryRegistries(true), 5000);
+        }
+
         this.setState({
           items: result.data.data,
           itemCount: result.data.meta.count,
           loading: false,
-        }),
-      ),
+        });
+      }),
     );
   }
 
-  private updateRegistries() {
-    ExecutionEnvironmentRegistryAPI.list(this.state.params).then((result) => {
-      const isAllCompleted = result.data.data.every(
-        (task) => task.last_sync_task.state === 'completed',
-      );
-
-      if (!isAllCompleted) setTimeout(() => this.updateRegistries(), 500);
-
-      this.setState({
-        items: result.data.data,
-        itemCount: result.data.meta.count,
-        loading: false,
-      });
-    });
-  }
-
   private deleteRegistry({ pk, name }) {
     ExecutionEnvironmentRegistryAPI.delete(pk)
       .then(() =>
@@ -495,7 +485,7 @@ class ExecutionEnvironmentRegistryList extends React.Component<
             </Trans>
           </span>,
         );
-        this.updateRegistries();
+        this.queryRegistries(true);
       })
       .catch(() => this.addAlert(t`Sync failed for ${name}`, 'danger'));
   }
```

File: src/containers/execution-environment/registry-list.tsx (hunk 2 above)
Comment (jerabekjiri) [reply src/containers/execution-environment/registry-list.tsx:456]: Cool overview of all timeouts, I'll use interval 5000 then, thanks :)

**Comment by jerabekjiri** [comment]
@himdel rewritten to your solution, much nicer and working good, thanks :+1:

**Comment by awcrosby** [comment]
(adding this to a few recently updated PRs…) We need CI in this repo that enforces this, but for now please:
* Add a line to at least one of your commits that says `Issue: AAH-####` (in the future it is best to be the first commit as it gets included in the initial PR description by default)
* Ensure when merging that the `Issue: AAH-####` is part of commit message via the “Squash ‘n’ Merge” description

**Comment by jerabekjiri** [comment]
Added `Issue: AAH-1094` to the latest commit description

**Review by himdel — APPROVED** [review]

**Pull request merged**